flask_praetorian/base.py

```
@@ -721,21 +721,29 @@ def read_token_from_cookie(self):
         return self._unpack_cookie(flask.request.cookies)
 
     def read_token(self):
-        exc = None
-        if 'header' in self.jwt_places:
-            try:
-                return self.read_token_from_header()
-            except MissingToken as e:
-                exc = e
-        if 'cookie' in self.jwt_places:
+        """
+        Tries to unpack the token from the current flask request in the locations configured with JWT_PLACES
+        Check-Order is defined by the value order in JWT_PLACES.
+        """
+        token=None
+        error=None
+        for place in self.jwt_places:
             try:
-                return self.read_token_from_cookie()
-            except MissingToken as e:
-                exc = e
-        if exc:
-            raise MissingToken("JWT token not found in {}".format(
-                self.jwt_places))
-
+                token = getattr(self, "read_token_from_{place}".format(place=place.lower()))()
+                if token:
+                    break
+            except MissingToken as err:
+                error=err
+            except AttributeError:
+                flask.current_app.logger.warning("Flask_Praetorian hasn't implemented reading JWT tokens from location {place}. Please reconfigure JWT_PLACES.".format(place=place.lower()))
+                flask.current_app.logger.warning("Values accepted in JWT_PLACES are: {def_places}".format(def_places=DEFAULT_JWT_PLACES))
+
+        if not token and error:
```

I think it's sufficient here to just check for token to be set. If token is falsey, this should throw the MissingToken exception.